File: CODE/DART/arc_backward_vector_type.dart
// -- TYPES

enum ArcBackwardVectorType
{
    // -- CONSTANTS

    custom( 0 ),
    symmetrical( 1 ),
    residual( 2 ),
    count( 3 ),
    none( -1 );

    // -- ATTRIBUTES
    
    final int 
        value;

    // -- CONSTRUCTORS

    const ArcBackwardVectorType(
        this.value
        );
}

File: CODE/RUST/src/arc_backward_vector_type.rs
// -- TYPES

#[derive( Debug, Clone, Copy, PartialEq )]
pub enum ArcBackwardVectorType
{
    // -- CONSTANTS
    
    CUSTOM = 0,
    SYMMETRICAL = 1,
    RESIDUAL = 2,
    COUNT = 3,
    NONE = -1
}

